### PR TITLE
First draft of analyzers

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak

File: tools/KodKod.Analyzer/KodKod.Analyzer/ApiActionsAreAttributeRoutedAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace KodKod.Analyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ApiActionsAreAttributeRoutedAnalyzer : KodKodAnalyzerAnalyzerBase
```
Comment (pranavkm) [inline tools/KodKod.Analyzer/KodKod.Analyzer/ApiActionsAreAttributeRoutedAnalyzer.cs:8]: Presumably we could put a couple of fixers - that recommend decorating it with `HttpGet` \ `HttpPost`.

**Comment by rynowak** [comment]
DANK

File: tools/KodKod.Analyzer/KodKod.Analyzer/ApiActionsDoNotRequireExplicitModelValidationCheckAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            kodKodContext.Context.RegisterSyntaxNodeAction(context =>
+            {
+                var methodSyntax = (MethodDeclarationSyntax)context.Node;
+                var method = context.SemanticModel.GetDeclaredSymbol(methodSyntax, context.CancellationToken);
+                if (!kodKodContext.IsKodKod(method))
+                {
+                    return;
+                }
+
+                if (method.ReturnsVoid || method.ReturnType == kodKodContext.SystemThreadingTask)
```
Comment (pranavkm) [inline tools/KodKod.Analyzer/KodKod.Analyzer/ApiActionsDoNotRequireExplicitModelValidationCheckAnalyzer.cs:28]: Need to do this better to support arbitrary awaitables.

**Comment by pranavkm** [comment]
bored

**Pull request closed without merging**